src/main/java/com/gregtechceu/gtceu/api/machine/trait/RecipeLogic.java

```diff
@@ -327,12 +322,24 @@ public void setupRecipe(GTRecipe recipe) {
         }
     }
 
+    private void notifyRecipeStatusChanged(Status oldStatus, Status newStatus) {
+        if (lastRecipe == null) {
+            return;
+        }
+        if (oldStatus == Status.WORKING && newStatus != Status.WORKING) {
+            lastRecipe.postWorking(machine);
```

Comment:
what if `newStatus == Status.WAITING`? then the machine will do recipe completed actions whilst it's just out of energy.

Reply:
other cases like this exist as well.